### PR TITLE
[SDK/CLI] Add chat group schema and node entity

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

Add chat group schema and node entity in experiment.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/test_configs/flows/chat_group_eval_history/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Basic Eval
```
Comment (wangchao1230) [inline src/promptflow/tests/test_configs/flows/chat_group_eval_history/README.md:1]: this file seems no use?

File: src/promptflow/promptflow/_sdk/entities/_experiment.py [hunk 2]
```diff
@@ -188,6 +189,46 @@ def _save_snapshot(self, target):
...
+        logger.debug(f"Saving chat group node {self.name!r} snapshot to {target.as_posix()!r}.")
+        saved_path = target / self.name
+        saved_path.mkdir(parents=True, exist_ok=True)
+        for role in self.roles:
+            role_path = Path(role["path"]).resolve()
+            if not role_path.exists():
+                raise ExperimentValueError(f"Chat role path {role_path.as_posix()!r} does not exist.")
+
+            if role_path.is_dir():
+                shutil.copytree(src=role_path, dst=saved_path / role["role"])
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_experiment.py:227]: do we have some common utility here? shall we handle advanced feature like addtional_includes here?

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/tests/test_configs/flows/chat_group_eval_history/README.md (hunk 1 above)
Comment (0mza987) [reply src/promptflow/tests/test_configs/flows/chat_group_eval_history/README.md:1]: copied from the example repo, could remove it later.

File: src/promptflow/promptflow/_sdk/entities/_experiment.py (hunk 2 above)
Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_experiment.py:227]: Think we can do it once we have multiple places need this functionality. For additional includes ask, good point i have created a task to track.

**Review by brynn-code — APPROVED** [review]

**Pull request merged**